com.cinecred.cinecred.yml

```
@@ -0,0 +1,41 @@
+id: com.cinecred.cinecred
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+command: cinecred
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --filesystem=host
+modules:
+  - name: ghostscript
+    config-opts:
+      - --disable-cups
+      - --disable-dbus
+      - --disable-gtk
+    cleanup:
+      - /share/doc
+      - /share/man
+    sources:
+      - type: archive
+        url: https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs10040/ghostscript-10.04.0.tar.gz
+        sha256: c764dfbb7b13fc71a7a05c634e014f9bb1fb83b899fe39efc0b6c3522a9998b1
+  - name: cinecred
+    buildsystem: simple
+    build-commands:
+      - sed -i "s/Icon=cinecred/Icon=$FLATPAK_ID/" cinecred.desktop
+      - sed -i "s/>cinecred/>$FLATPAK_ID/" cinecred.metainfo.xml
+      - mkdir -p /app/share/{applications/,metainfo/} /app/share/icons/hicolor/{scalable/apps/,256x256/apps/}
+      - mv cinecred.desktop "/app/share/applications/$FLATPAK_ID.desktop"
+      - mv cinecred.metainfo.xml "/app/share/metainfo/$FLATPAK_ID.metainfo.xml"
+      - mv cinecred.svg "/app/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg"
+      - mv cinecred.png "/app/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png"
```

```suggestion
      - install -Dm644 cinecred.desktop "/app/share/applications/$FLATPAK_ID.desktop"
      - install -Dm644 cinecred.metainfo.xml "/app/share/metainfo/$FLATPAK_ID.metainfo.xml"
      - install -Dm644 cinecred.svg "/app/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg"
      - install -Dm644 cinecred.png "/app/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png"
```

nonwithstanding installing them with the original names as mentionned above.

---

I now use the `rename-*` properties and the `install` command; this should also resolve the two previous comments. One `sed` is however still needed to delete the `<icon>` tag in the AppStream XML file, because that tag not automatically renamed by `flatpak-builder`, but instead later amended by the "Composing metadata..." step.

---

no, `sed` is not necessary. and in any case patches should be used as `sed` failure mode is...  a failure.

Ideally you'll just use the naming across the board, it's not flatpak specific.

---

When I build the Flatpak locally without the remaining `sed` command, I get the following error:

```
Renaming cinecred.metainfo.xml to share/metainfo/com.cinecred.cinecred.metainfo.xml
Renaming cinecred.desktop to com.cinecred.cinecred.desktop
Renaming icon hicolor/256x256/apps/cinecred.png to hicolor/256x256/apps/com.cinecred.cinecred.png
Renaming icon hicolor/scalable/apps/cinecred.svg to hicolor/scalable/apps/com.cinecred.cinecred.svg
Rewriting contents of com.cinecred.cinecred.desktop
Running appstreamcli compose
Only accepting components: com.cinecred.cinecred, com.cinecred.cinecred.desktop
Processing directory: .../flathub/.flatpak-builder/rofiles/rofiles-MgpDk2/files
Composing metadata...
Run failed, some data was ignored.
Errors were raised during this compose run:
general
  E: filters-but-no-output

com.cinecred.cinecred
  E: icon-not-found
Refer to the generated issue report data for details on the individual problems.
Error: ERROR: appstreamcli compose failed: Der Kindprozess wurde mit Status 1 beendet
```

The cause is that the upstream `cinecred.metainfo.xml` contains the tag `<icon type="stock">cinecred</icon>`, and `flatpak-builder` doesn't rename it, so deleting it via `sed` is the simplest solution. If this `sed` call failed, the whole Flatpak build would fail, as is evident from the error log above. So in this case, the failure condition of `sed` is to make the whole build fail, which I think is right.